network/proto/impl.go

```diff
@@ -111,6 +133,7 @@ func (p *protocol) Start() {
 	go p.startUpdatingDiagnostics(peerConnected, peerDisconnected)
 	go p.startAdvertingHashes()
 	go p.startAdvertingDiagnostics()
+	go p.startCollectingMissingPayloads()
```

Comment:
I'm wondering. What's the strategy for shutting down?

Reply:
Glad you asked, I'm wondering as well. Not where it is (because I simply did't write it) but why it seems to work fine without it.

As goroutines are shut down together with the process, the program *will* exit some way or another. Since we don't write to (local) resources from these goroutines, I think nothing will be corrupted by the lacking shutdown mechanic. But it isn't very neat either, because it may cause the occasional weird error message when shutting down the Nuts node.

Is it OK to create an issue to fix it for all goroutines in this engine, rather than fixing it for this specific one?

Reply:
I guess it's fine-ish as there shouldn't be any corruption indeed but like you said it would be nice to shut it down 'properly'.

> Is it OK to create an issue to fix it for all goroutines in this engine, rather than fixing it for this specific one?

Yeah, makes sense 🙂

Reply:
https://github.com/nuts-foundation/nuts-node/issues/480